gphoto2cffi/backend.py

```diff
@@ -131,6 +123,22 @@ def __getattr__(self, name):
         else:
             return val
 
+    def _logging_callback(self, level, domain, message, data):
```

Comment:
Have you tried getting rid of the wrapped inner function with 2)? I'm not sure how cffi handles callbacks that are bound instance methods, but that would probably the cleanest way from a readability perspective. Otherwise I'd probably go with 3) :-)